lib/adagrams.rb

```diff
@@ -0,0 +1,97 @@
+require "csv"
+require "set"
+
+def draw_letters
+  letters_hash = {
+    "A" => 9, "B" => 2, "C" => 2, "D" => 4, "E" => 12, "F" => 2,
+    "G" => 3, "H" => 2, "I" => 9, "J" => 1, "K" => 1, "L" => 4,
+    "M" => 2, "N" => 6, "O" => 8, "P" => 2, "Q" => 1, "R" => 6,
+    "S" => 4, "T" => 6, "U" => 4, "V" => 2, "W" => 2, "X" => 1,
+    "Y" => 2, "Z" => 1,
+  }
+
+  letters = []
+  letters_hash.each { |letter, frequency|
+    frequency.times do
+      letters << letter
+    end
+  }
+
+  return letters.sample(10)
```

Comment:
Clever! Nicely done to use the optional argument of `sample` to get exactly what you need in one line.